src/Type/Php/JsonValidateTypeSpecifyingExtension.php

```diff
@@ -0,0 +1,42 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Analyser\SpecifiedTypes;
+use PHPStan\Analyser\TypeSpecifier;
+use PHPStan\Analyser\TypeSpecifierAwareExtension;
+use PHPStan\Analyser\TypeSpecifierContext;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\Accessory\AccessoryNonFalsyStringType;
+use PHPStan\Type\FunctionTypeSpecifyingExtension;
+use function count;
+
+class JsonValidateTypeSpecifyingExtension implements FunctionTypeSpecifyingExtension, TypeSpecifierAwareExtension
+{
+
+	private TypeSpecifier $typeSpecifier;
+
+	public function isFunctionSupported(FunctionReflection $functionReflection, FuncCall $node, TypeSpecifierContext $context): bool
+	{
+		return $functionReflection->getName() === 'json_validate' && isset($node->getArgs()[0]) && $context->true();
+	}
+
+	public function specifyTypes(FunctionReflection $functionReflection, FuncCall $node, Scope $scope, TypeSpecifierContext $context): SpecifiedTypes
+	{
+		$args = $node->getArgs();
+
+		if (count($args) < 1) {
+			return new SpecifiedTypes([], []);
+		}
+
+		return $this->typeSpecifier->create($node->getArgs()[0]->value, new AccessoryNonFalsyStringType(), $context, false, $scope);
```

Comment:
is non falsy string type correct here?

https://3v4l.org/5IGU2/rfc#vgit.master

`'0'` is a valid json string but is also falsey

Reply:
You are right. I did a 3v4l before with this wxact case in mind. Seems I fucked it up. 

Need to go with non-empty-string then :-)